### PR TITLE
Added doc for maximum number formed from the given numbers in an array

### PR DESCRIPTION
**Pull request by SAY-droid427**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes https://github.com/TesseractCoding/NeoAlgo-Docs/issues/36

### CODE REVIEW

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Please make changes in the `sidebars.js`on the root. Otherwise it would be impossible to review the article on a live preview.

**Comment by SAY-droid427** [comment]
@HarshCasper Please review

**Comment by SAY-droid427** [comment]
@HarshCasper Please review

**Comment by SAY-droid427** [comment]
@HarshCasper The changes you had asked me to make in the other PRs, like using backticks and converting to heading 2, I have chaged here also

File: docs/competitive-programming/max-number-from-array.mdx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+slug: /competitive-programming/max-number-from-array
+---
+
+## Introduction
+
+To form the maximum number possible from an array of single-digit numbers, we have to sort the array in desscending order. The numbers appended in the descending order gives the maximum number.
+However, the same algorithm cannot be used for array containing both two-digit and single-digit numbers. 
+For example, if the array contains two elements, 2 and 21, then according to the algorithm used for the single digits, if we sort the array in descending order and append them, then the maximum number will be 212.
+But this is wrong. The maximum number that can be formed from this array is 221.
+Therefore, we need to use a separate algorithm.
```
Comment (HarshCasper) [inline docs/competitive-programming/max-number-from-array.mdx:12]: Wrap all of this in one line. Use backticks with numbers. Fix the typos

File: docs/competitive-programming/max-number-from-array.mdx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+## Algorithm
+
+1. Input the numbers as string
+2. Sort the array of strings based on the following comparison criteria
+    * While sorting the array using comparison, create two separate strings XY and YX, if X and Y are the two strings being compared
+    * XY is the string created by appending X and Y in order
+    * YX is the string created by appending Y and X in order.
+    * Compare XY and YX and whichever is greater, should be placed before the other in the array
```
Comment (HarshCasper) [inline docs/competitive-programming/max-number-from-array.mdx:21]: Use bulleted list not ordered list

File: docs/competitive-programming/max-number-from-array.mdx [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+## Examples
+
+### Example 1
```
Comment (HarshCasper) [inline docs/competitive-programming/max-number-from-array.mdx:25]: Use Heading-4

File: docs/competitive-programming/max-number-from-array.mdx [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+```
+Let the array be {21,2}
+While sorting, two strings are created : XY = 212 and YZ = 221
+As XY<YZ, therefore, the array is sorted as {2,21}
+Therefore, the maximum number is 221
+```
+
+### Example 2
```
Comment (HarshCasper) [inline docs/competitive-programming/max-number-from-array.mdx:34]: Use Heading-4

File: docs/competitive-programming/max-number-from-array.mdx [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+</TabItem>
+</Tabs>
+
+## Sample Input and Output
+
+#### Input:
+Enter the number of elements in array
+3
+Enter the elements
+10 21 34
```
Comment (HarshCasper) [inline docs/competitive-programming/max-number-from-array.mdx:99]: Keep in triple backticks

File: docs/competitive-programming/max-number-from-array.mdx [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+#### Output:
+The maximum number : 342110
+
+
+#### Input:
+Enter the number of elements in array
+4
+Enter the elements
+12 21 33 26
```
Comment (HarshCasper) [inline docs/competitive-programming/max-number-from-array.mdx:109]: Keep in triple backticks

File: docs/competitive-programming/max-number-from-array.mdx [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+## Sample Input and Output
+
+#### Input:
+Enter the number of elements in array
+3
+Enter the elements
+10 21 34
+
+#### Output:
+The maximum number : 342110
```
Comment (HarshCasper) [inline docs/competitive-programming/max-number-from-array.mdx:102]: Keep in triple backticks

File: docs/competitive-programming/max-number-from-array.mdx [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+
+
+#### Input:
+Enter the number of elements in array
+4
+Enter the elements
+12 21 33 26
+
+#### Output:
+The maximum number : 33262112
```
Comment (HarshCasper) [inline docs/competitive-programming/max-number-from-array.mdx:112]: Keep in triple backticks

File: docs/competitive-programming/max-number-from-array.mdx [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+
+## Complexity Analysis
+
+* Time Complexity: `O(n)`
+* Space Complexity : `O(1)`
+
+## Credit
+
+[Sayani Mallick](https://github.com/SAY-droid427/) for the [C++ implementation](https://github.com/TesseractCoding/NeoAlgo/blob/master/C-Plus-Plus/cp/maxNumber.cpp)
```
Comment (HarshCasper) [inline docs/competitive-programming/max-number-from-array.mdx:121]: Keep a bullet point

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Please resolve the merge conflicts as well

File: docs/competitive-programming/max-number-from-array.mdx (hunk 1 above)
Comment (SAY-droid427) [reply docs/competitive-programming/max-number-from-array.mdx]: Done

File: docs/competitive-programming/max-number-from-array.mdx (hunk 2 above)
Comment (SAY-droid427) [reply docs/competitive-programming/max-number-from-array.mdx]: Done

File: docs/competitive-programming/max-number-from-array.mdx (hunk 3 above)
Comment (SAY-droid427) [reply docs/competitive-programming/max-number-from-array.mdx]: Done

File: docs/competitive-programming/max-number-from-array.mdx (hunk 4 above)
Comment (SAY-droid427) [reply docs/competitive-programming/max-number-from-array.mdx]: Done

File: docs/competitive-programming/max-number-from-array.mdx (hunk 5 above)
Comment (SAY-droid427) [reply docs/competitive-programming/max-number-from-array.mdx:96]: Done

File: docs/competitive-programming/max-number-from-array.mdx (hunk 6 above)
Comment (SAY-droid427) [reply docs/competitive-programming/max-number-from-array.mdx:107]: Done

File: docs/competitive-programming/max-number-from-array.mdx (hunk 8 above)
Comment (SAY-droid427) [reply docs/competitive-programming/max-number-from-array.mdx:110]: Done

File: docs/competitive-programming/max-number-from-array.mdx (hunk 9 above)
Comment (SAY-droid427) [reply docs/competitive-programming/max-number-from-array.mdx]: Done

File: sidebars.js [hunk 10]
```diff
@@ -1,4 +1,152 @@
...
+        'thank-you',
+      ],
+    },
+  ],
+};
+module.exports = {	
+  
     docs: [
         {
             type: "category",
```
Comment (HarshCasper) [inline sidebars.js:152]: This is repeated twice in the file now

File: docs/competitive-programming/max-number-from-array.mdx [hunk 11]
```diff
@@ -0,0 +1,120 @@
+---
+title: Maximum number formed from numbers in array
+slug: /competitive-programming/max-number-from-array
+---
+
+## Introduction
+
+To form the maximum number possible from an array of single-digit numbers, we have to sort the array in descending order. The numbers appended in the descending order give the maximum number. However, the same algorithm cannot be used for array containing both two-digit and single-digit numbers.For example, if the array contains two elements, `2` and `21`, then according to the algorithm used for the single digits, if we sort the array in descending order and append them, then the maximum number will be `212`.But this is wrong. The maximum number that can be formed from this array is `221`.Therefore, we need to use a separate algorithm.
```
Comment (HarshCasper) [inline docs/competitive-programming/max-number-from-array.mdx:8]: Keep proper spaces after full-stops

File: docs/competitive-programming/max-number-from-array.mdx [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+
+## Algorithm
+
+- Input the numbers as string
+- Sort the array of strings based on the following comparison criteria
+    * While sorting the array using comparison, create two separate strings XY and YX, if X and Y are the two strings being compared
+    * XY is the string created by appending X and Y in order
+    * YX is the string created by appending Y and X in order.
+    * Compare XY and YX and whichever is greater, should be placed before the other in the array
```
Comment (HarshCasper) [inline docs/competitive-programming/max-number-from-array.mdx:17]: Put full stops everywhere

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: docs/competitive-programming/max-number-from-array.mdx (hunk 12 above)
Comment (SAY-droid427) [reply docs/competitive-programming/max-number-from-array.mdx]: Done

File: docs/competitive-programming/max-number-from-array.mdx (hunk 11 above)
Comment (SAY-droid427) [reply docs/competitive-programming/max-number-from-array.mdx]: Done

File: sidebars.js (hunk 10 above)
Comment (SAY-droid427) [reply sidebars.js]: Done

File: docs/competitive-programming/max-number-from-array.mdx [hunk 13]
```diff
@@ -0,0 +1,120 @@
+---
+title: Maximum number formed from numbers in array
```
Comment (HarshCasper) [inline docs/competitive-programming/max-number-from-array.mdx:2]: `Maximum number formed from numbers in an array`

File: docs/competitive-programming/max-number-from-array.mdx [hunk 14]
```diff
@@ -0,0 +1,120 @@
...
+slug: /competitive-programming/max-number-from-array
+---
+
+## Introduction
+
+To form the maximum number possible from an array of single-digit numbers, we have to sort the array in descending order. The numbers appended in the descending order give the maximum number. However, the same algorithm cannot be used for array containing both two-digit and single-digit numbers. For example, if the array contains two elements, `2` and `21`, then according to the algorithm used for the single digits, if we sort the array in descending order and append them, then the maximum number will be `212`. But this is wrong. The maximum number that can be formed from this array is `221`. Therefore, we need to use a separate algorithm.
```
Comment (HarshCasper) [inline docs/competitive-programming/max-number-from-array.mdx:8]: First introduce the problem. What exactly we are trying to do here and what is its relevance. Some of the part can be shifted to the explanation

File: docs/competitive-programming/max-number-from-array.mdx [hunk 15]
```diff
@@ -0,0 +1,120 @@
...
+
+## Sample Input and Output
+
+#### Input:
+```
+Enter the number of elements in array:
+3
+Enter the elements:
+10 21 34
+
```
Comment (HarshCasper) [inline docs/competitive-programming/max-number-from-array.mdx:97]: Keep triple backticks here and put a new line

File: docs/competitive-programming/max-number-from-array.mdx [hunk 16]
```diff
@@ -0,0 +1,120 @@
...
+#### Output:
```
Comment (HarshCasper) [inline docs/competitive-programming/max-number-from-array.mdx:98]: Keep a space after this heading

File: docs/competitive-programming/max-number-from-array.mdx [hunk 17]
```diff
@@ -0,0 +1,120 @@
...
+The maximum number : 342110
+```
+
+```
+#### Input:
+Enter the number of elements in array:
+4
+Enter the elements:
+12 21 33 26
+
```
Comment (HarshCasper) [inline docs/competitive-programming/max-number-from-array.mdx:108]: Keep triple backticks here and put a new line

File: docs/competitive-programming/max-number-from-array.mdx [hunk 18]
```diff
@@ -0,0 +1,120 @@
...
+#### Output:
```
Comment (HarshCasper) [inline docs/competitive-programming/max-number-from-array.mdx:109]: Keep a space after this heading

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: docs/competitive-programming/max-number-from-array.mdx (hunk 13 above)
Comment (SAY-droid427) [reply docs/competitive-programming/max-number-from-array.mdx]: Done

File: docs/competitive-programming/max-number-from-array.mdx (hunk 14 above)
Comment (SAY-droid427) [reply docs/competitive-programming/max-number-from-array.mdx]: Done. I have written a small introduction to the problem and moved the example to the problem in the Explanation section stating why another algorithm is needed.

File: docs/competitive-programming/max-number-from-array.mdx (hunk 15 above)
Comment (SAY-droid427) [reply docs/competitive-programming/max-number-from-array.mdx:105]: Done

File: docs/competitive-programming/max-number-from-array.mdx (hunk 16 above)
Comment (SAY-droid427) [reply docs/competitive-programming/max-number-from-array.mdx]: Done

File: docs/competitive-programming/max-number-from-array.mdx (hunk 18 above)
Comment (SAY-droid427) [reply docs/competitive-programming/max-number-from-array.mdx]: Done

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**